advanced/Scripts/piholeDebug.sh

```diff
@@ -396,49 +396,46 @@ check_critical_program_versions() {
 os_check() {
     # This function gets a list of supported OS versions from a TXT record at versions.pi-hole.net
     # and determines whether or not the script is running on one of those systems
-    local remote_os_domain valid_os valid_version detected_os_pretty detected_os detected_version
+    local remote_os_domain valid_os valid_version detected_os detected_version
     remote_os_domain="versions.pi-hole.net"
-    valid_os=false
-    valid_version=false
 
-    detected_os_pretty=$(cat /etc/*release | grep PRETTY_NAME | cut -d '=' -f2- | tr -d '"')
-    detected_os="${detected_os_pretty%% *}"
-    detected_version=$(cat /etc/*release | grep VERSION_ID | cut -d '=' -f2- | tr -d '"')
+    detected_os=$(grep "\bID\b" /etc/os-release | cut -d '=' -f2 | tr -d '"')
```

Comment:
I wonder if we should add in a check for the existence of `/etc/os-release`, and if it is not there, show a message saying we are unable to detect the OS (similarly, if we get a blank response on the `dig` command?)

Reply:
Sure, if you do `grep -s` then you won't get the "file not found" error and the output will be empty.

```plain
dan@Pi-hole:~$ det=$(grep "\bID\b" /etc/os-release)
dan@Pi-hole:~$ echo $det
ID=debian
dan@Pi-hole:~$ det=$(grep -s "\bID\b" /etc/no-release)
dan@Pi-hole:~$ echo $det

dan@Pi-hole:~$
```

Reply:
```plain
dan@Pi-hole:~$ det=$(grep "\bID\b" /etc/no-release)
grep: /etc/no-release: No such file or directory
dan@Pi-hole:~$ echo $det

dan@Pi-hole:~$
```

Reply:
#3695 addresses the `dig`, may want to comment on that.

Reply:
I'll give it a whirl in a bit, It's probably going to need a rebase, that said.

Reply:
Sounds good, I'll resolve this convo then and it can be taken up on #3695